### ISSUE TITLE
Absl flags pathlib support

### ISSUE THREAD

**Issue by brycearden** [abseil/abseil-py#165]
Are there any plans to support a `flags.DEFINE_path` or something similar that utilizes python's `pathlib` module to support declaring `os.PathLike` operations for the default.

For context, I often find myself doing the following:

```py
from absl import app, flags
from pathlib import Path

FLAGS = flags.FLAGS
flags.DEFINE_string("filepath", "path/to/some.txt", "Input filepath for program")

def main(argv_):
    # Override FLAGS.filepath with a pathlib.Path version
    FLAGS.filepath = Path(FLAGS.filepath)
    
    # Rest of program uses pathlib for path operations

if __name__ == "__main__":
    app.run(main)
```

It would be awesome if there was an easy way to define a `Path` object from the `DEFINE_` definition in the first place. If adding such a feature is not feasible, I'm curious how other people are using `pathlib` with `absl.flags` as well.

Thanks!

**Comment by gpshead** [comment]
Internally I do see one recent user contributed library that implements its own `DEFINE_path` for use with `absl`.  The gist of that goes something like this _(simplified for illustration purposes)_ in a`path_flag.py` library:

```python
flags.disclaim_key_flags()  # Prevent absl from attributing flags to this module.

class _PathParser(flags.ArgumentParser):
    def parse(self, value):
        return pathlib.PurePath(value)

class _PathSerializer(flags.ArgumentSerializer):
    def serialize(self, value):
        return str(value)

def DEFINE_path(...):
    return flags.DEFINE(_PathParser(), name, default, help, flag_values,
                        _PathSerializer(), **kwargs)
```

I agree, it'd make sense to add something like that to absl itself as pathlib is gaining popularity.

_Internal source: `pyglib.contrib.gpathlib` - If we do this, presumably we might just refactor it out of that._

**Comment by danielkelshaw** [comment]
Just wondering if there was any update on this? I'd say the `pathlib` library is fairly standard now so it would be nice to have support from `absl` if possible.

**Comment by yilei** [comment]
Yes this is in my queue, hopefully I can get to it after the holidays if not earlier.

**Comment by aryamccarthy** [comment]
Perhaps `DEFINE_path` can happen by these holidays? :-)

**Comment by markmcd** [comment]
[`etils`](https://pypi.org/project/etils) has `DEFINE_path`. It returns a `pathlib` object. See the [`etils.epath`](https://etils.readthedocs.io/en/latest/epath.html) docs.

```py
_PATH = epath.DEFINE_path('path', None, 'Path to input file.')
```

This is a Googler-project, so may be what gps was describing. Or maybe not :)
